File: code_c/calckeyb.c
#include "parm.h"
#include "utils.h"
#include "math.h"

void run()
{
	BEGIN();

	int a, b;

	PUTCHAR('A','=');
	a = READINT();
	PUTCHAR('\n');

	PUTCHAR('B','=');
	b = READINT();
	PUTCHAR('\n');

	PUTCHAR('+','-','*','/','%','&','|','^');

	int choice;
read:
	choice = READKEY();

	switch(choice) 
	{
	case '+':
		RES = a + b;
		break;
	case '-':
		RES = a - b;
		break;
	case '*':
		RES = a * b;
		break;
	case '/':
		RES = DIV(a, b);
		break;
	case '%':
		RES = MOD(a, b);
		break;
	case '&':
		RES = a & b;
		break;
	case '|':
		RES = a | b;
		break;
	case '^':
		RES = a ^ b;
		break;
	default:
		goto read;
	}

	PUTCHAR('\n','R','=');
	PRINTRES_SIGN();

	WAITKEY();

	RESET();
}

File: code_rs/test/src/calckeyb.rs
#![feature(min_specialization)]
#![feature(step_trait)]
#![no_main]
#![no_std]

use crate::parm::keyb::read_key;

use crate::parm::mmio::RES;
use crate::parm::tty::read_int;

mod parm;

fn main() {
    'main:
    loop {
        print!("A = ");
        let a = read_int();

        print!("B = ");
        let b = read_int();

        println!("+-*/%&|^");
        loop {
            let choice = read_key();
            let res = match choice as u8 {
                b'+' => a + b,
                b'-' => a - b,
                b'*' => a * b,
                b'/' => a / b,
                //b'%' => math::r#mod(a, b),
                b'&' => a & b,
                b'|' => a | b,
                b'^' => a ^ b,
                b'\n' => continue 'main,
                _ => continue,
            };
            RES.write(res);
        }
    }
}
